src/System.Management.Automation/engine/MshCommandRuntime.cs

```diff
@@ -3250,8 +3250,7 @@ internal SwitchParameter WhatIf
             {
                 if (!IsWhatIfFlagSet && !_isWhatIfPreferenceCached)
                 {
-                    bool defaultUsed = false;
-                    _whatIfFlag = Context.GetBooleanPreference(SpecialVariables.WhatIfPreferenceVarPath, _whatIfFlag, out defaultUsed);
+                    _whatIfFlag = Context.GetBooleanPreference(SpecialVariables.WhatIfPreferenceVarPath, _whatIfFlag, out _);
```

Comment:
Thank you. It bothered me that you could assign anything to `$WhatIfPrefrence` and that value stuck:
```
PS> $WhatIfPreference = Get-Date
PS> $WhatIfPreference

Wednesday, August 25, 2021 10:53:41 AM
```